kart/raster/v1.py

```diff
@@ -237,11 +245,47 @@ def _non_pam_wc_paths(self, tile_and_pam_paths):
                 tile_paths = find_similar_files_case_insensitive(
                     self._workdir_path(path[:-LEN_PAM_SUFFIX])
                 )
-                result.update("/".join(t.relative_to(wc_path).parts) for t in tile_paths)
+                result.update(
+                    "/".join(t.relative_to(wc_path).parts) for t in tile_paths
+                )
             else:
                 result.add(path)
         return result
 
+    def _only_stats_have_changed(self, tile_delta):
+        """
+        Given a tile delta where the new value is the current state of the WC,
+        return True if the only thing that has changed in the tile since the last commit
+        is that stats have been added or removed.
+        """
+        old_value = tile_delta.old_value
+        new_value = tile_delta.new_value
+
+        all_keys = set()
+        all_keys.update(old_value)
+        all_keys.update(new_value)
+        for key in all_keys:
+            if not key.startswith("pam") and old_value.get(key) != new_value.get(key):
+                return False
+
+        old_pam_oid = old_value.get("pamOid")
+        new_pam_oid = new_value.get("pamOid")
+        if old_pam_oid == new_pam_oid:
+            return True
+        if old_pam_oid is None or new_pam_oid is None:
+            return False
```

Comment:
this is where the PAM is created/deleted?
if there's only stats in the PAM file, then maybe this should return True?

Reply:
Done